### PR TITLE
Release notes for 4.5.1

### PR DESCRIPTION
**Pull request by eolivelli**
This patch contains new site docs for 4.5.1

### CODE REVIEW

**Comment by eolivelli** [comment]
@sijie this is my proposal can you check please ?

**Comment by jvrao** [comment]
LGTM

File: site/docs/4.5.1/overview/releaseNotes.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: Apache BookKeeper 4.5.0 Release Notes
```
Comment (sijie) [inline site/docs/4.5.1/overview/releaseNotes.md]: 4.5.1?

File: site/docs/4.5.1/overview/releaseNotes.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+This is the sixth release of BookKeeper as an Apache Top Level Project!
+
+The 4.5.1 release is a bugfix release which fixes a bunch of issues found from users of 4.5.0.
+
+Apache BookKeeper users are encouraged to upgrade to 4.5.1. The technical details of this release are summarized
+below.
+
+## Highlights
+
+TODO
```
Comment (sijie) [inline site/docs/4.5.1/overview/releaseNotes.md]: no highlights? maybe you can just simply list some bug fixes?

File: site/docs/4.5.1/admin/upgrade.md [hunk 3]
```diff
@@ -59,15 +59,6 @@ For each Bookie:
...
-
-1. 4.5.x upgrades netty from 3.x to 4.x. The memory usage pattern might be changed a bit. Netty 4 uses more direct memory. Please pay attention to your memory usage
-    and adjust the JVM settings accordingly.
-2. `multi journals` is a non-rollbackable feature. If you configure a bookie to use multiple journals on 4.5.x you can not roll the bookie back to use 4.4.x. You have
-    to take a bookie out and recover it if you want to rollback to 4.4.x.
-
-If you are planning to upgrade a non-secured cluster to a secured cluster enabling security features in 4.5.0, please read [BookKeeper Security](../../security/overview) for more details.
+### 4.5.0 to 4.5.1 upgrade
 
+There isn't any protocol related backward compabilities changes in 4.5.0. So you can follow the general upgrade sequence to upgrade from 4.5.0 to 4.5.1.
```
Comment (ivankelly) [inline site/docs/4.5.1/admin/upgrade.md]: There isn't any incompatible protocol changes in 4.5.0.

File: site/docs/4.5.1/overview/releaseNotes.md [hunk 4]
```diff
@@ -15,487 +15,10 @@ TODO
 
 ### Dependencies Upgrade
```
Comment (ivankelly) [inline site/docs/4.5.1/overview/releaseNotes.md]: there must be something interesting in 4.5.1, otherwise why do the release?

**Comment by eolivelli** [comment]
@sijie We have two commits:
https://github.com/apache/bookkeeper/pull/678/commits/6526f47f9a09eae5284efa72a9f03f3d316a0329

https://github.com/apache/bookkeeper/pull/678/commits/685e6815b74f6e2bad3c126b8d678e92d47bdd19

I have rebased and re-split after addressing @ivankelly comments

I can't find how to generate the site locally
The first commit is the bare copy, you can check the second.
If you prefer I can send two pull requests

**Comment by sijie** [comment]
1) it is good to send two pull requests, simply because we can merge the first change very quickly, because it is a copy. the release notes requires more reviews.

2) regarding the copy, you might check `site/scripts/release.sh`. this script deals with major release (e.g. 4.5 -> 4.6). it doesn't deal with minor release. but you should create a similar one to deal with the minor release. because there are multiple places that you need to change the versions.

3) checkout `site/README.md`. it has the instructions to build the site locally and at staging.

**Comment by sijie** [comment]
I merged #680 . let's move forward with 4.5.1

**Comment by sijie** [comment]
@eolivelli website updates should be the last thing. did you check the process in release guide?

- cut the branch?
- release to staging?
- get everything (links to staging artifacts, release notes) ready for a rc voting email

**Comment by eolivelli** [comment]
I will do today.
Can you confirm that in this case we won't have a new branch. I will only create a tag

**Comment by sijie** [comment]
yes we only need to create a tag.

**Comment by jiazhai** [comment]
Since #680 is merged, It is better to rebase this.

File: site/releases.md [hunk 5]
```diff
@@ -27,6 +27,14 @@ Client Guide | API docs
 
 ## News
 
+### XX November, 2017: Release 4.5.1 available
```
Comment (jiazhai) [inline site/releases.md]: This may need change after the vote?

Comment (eolivelli) [reply site/releases.md]: Yes

**Comment by eolivelli** [comment]
@jiazhai I am going to rebase as soon as possible, thanks

**Comment by eolivelli** [comment]
@jiazhai rebased to latest master.

@sijie It is not clear to me what I have to do. The release script as you stated is good for major releases.
I still don't know the structure of the website. What changed should be pushed for a minor release ? Links on the menus ?
I think I did that some of the tasks of the script does, like creating the directory for 4.5.1

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

**Comment by sijie** [comment]
regarding the release script, it would be good to change the release.sh script to support both major release and minor release. since we already have a script, it would be good to improve the script for minor release, rather than do everything manually. it is going to be hard to a release manager to track every change should happen for a release. 

I am fine with you doing it in a separate task, but ideally we should have such a script.

File: site/docs/4.5.1/admin/upgrade.md [hunk 6]
```diff
@@ -59,15 +59,6 @@ For each Bookie:
 
 We describes the general upgrade method in Apache BookKeeper as above. We will cover the details for individual versions.
 
-### 4.4.x to 4.5.x upgrade
```
Comment (sijie) [inline site/docs/4.5.1/admin/upgrade.md]: can you keep 4.4.x to 4.5.x upgrade there? why do you need to remove it?

File: site/docs/4.5.1/overview/releaseNotes.md [hunk 7]
```diff
@@ -11,491 +11,16 @@ below.
 
 ## Highlights
 
-TODO
+- Ability to use only lookback interface on LocalBookie, see [https://github.com/apache/bookkeeper/issues/407](https://github.com/apache/bookkeeper/issues/407)
```
Comment (sijie) [inline site/docs/4.5.1/overview/releaseNotes.md]: don't think it worth a highlight. we can remove it.

File: site/docs/4.5.1/overview/releaseNotes.md [hunk 8]
```diff
@@ -11,491 +11,16 @@ below.
...
-
-- Upgraded netty from 3.x to 4.x to leverage buffer pooling and reduce memory copies.
-- Moved developement from Java 7 to Java 8 to take advantage of Java 8 features.
-- A lot of improvements around scheduling and threading on `bookies`.
-- Delay ensemble change to improve tail latency.
-- Parallel ledger recovery to improve the recovery speed.
-- ...
-
-We outlined following four changes as below. For a complete list of performance improvements, please checkout the `full list of changes` at the end.
+- Fix critical bug on Parallel Recovery, see [https://github.com/apache/bookkeeper/issues/343](https://github.com/apache/bookkeeper/issues/343)
```
Comment (sijie) [inline site/docs/4.5.1/overview/releaseNotes.md]: there are also a few others:

- ledger length in ledgerhandleadv
- rack aware placement policy

File: site/releases.md [hunk 9]
```diff
@@ -27,6 +27,14 @@ Client Guide | API docs
...
+
+This is the sixth release of BookKeeper as an Apache Top Level Project!
+
+The 4.5.1 release is a bugfix release.
```
Comment (ivankelly) [inline site/releases.md]: A quick summary of the highlights would be good here.

File: site/docs/4.5.1/admin/upgrade.md (hunk 6 above)
Comment (eolivelli) [reply site/docs/4.5.1/admin/upgrade.md]: done

File: site/docs/4.5.1/overview/releaseNotes.md (hunk 7 above)
Comment (eolivelli) [reply site/docs/4.5.1/overview/releaseNotes.md]: done

File: site/docs/4.5.1/overview/releaseNotes.md (hunk 8 above)
Comment (eolivelli) [reply site/docs/4.5.1/overview/releaseNotes.md]: done

File: site/releases.md (hunk 9 above)
Comment (eolivelli) [reply site/releases.md]: isn't "The technical details of this release are summarized below" enough ?

**Comment by eolivelli** [comment]
@sijie 

> I am fine with you doing it in a separate task, but ideally we should have such a script.

created this issue:
https://github.com/apache/bookkeeper/issues/705

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

**Comment by asfgit** [comment]
FAILURE
 
--none--

**Pull request closed without merging**